.github/workflows/build.yaml

```diff
@@ -25,6 +25,8 @@ jobs:
         go-version: 1.15.x
     - name: Set up kubebuilder
       uses: fluxcd/pkg/actions/kubebuilder@main
+    - name: Install libgit2
+      run: sudo apt-get install -y libgit2-dev
```

Comment:
This will not work as libgit2 is a couple of years old on apt. You can no longer build the binary in the VM, only in container, see https://github.com/fluxcd/source-controller/blob/main/.github/workflows/e2e.yaml#L31